server/http.go

```diff
@@ -176,3 +177,34 @@ func (p *Plugin) respondSpecialTemplate(w http.ResponseWriter, key string, statu
 	}
 	return status, nil
 }
+
+type roundtripper struct {
+	http.RoundTripper
+	limit ByteSize
+}
+
+func (rt roundtripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	resp, err := rt.RoundTripper.RoundTrip(r)
+	if err != nil || resp == nil || resp.Body == nil {
+		return resp, err
+	}
+	resp.Body = struct {
+		io.Reader
+		io.Closer
+	}{
+		Reader: io.LimitReader(resp.Body, int64(rt.limit)),
```

Comment:
Maybe worth logging a ticket to add an own `Read` method that would return a different user-friendly error when a premature EOF is encountered. LimitReader returns an EOF, usually in the middle of a JSON parse.